sharmine-s/acebook-airbenders
# Comments

- Implemented comments on the backend
- Created new comment model, added CreateComment and UpdatePost routes and methods

To test this PR on Postman:
1. Drop db and logout
2. Sign up, log in and create a post in the UI
3. In Compass, copy the id of the created post
4. In Postman, POST to http://localhost:8080/posts/comments
5. In the body (raw, as JSON) - { "postId": "<your post id>", "message": "i am comment" }
6. Grab the token from the browser and add it in Headers - Authorization: bearer <token>
7. SEND. (You have now created a comment in a new comments collection in your db)
8. Now, to update the post to add the comment:
9.  Copy the id from your new comment (in compass)
10. In Postman, PATCH to http://localhost:8080/posts/comments
11. In the body (raw, as JSON) - {"postId": "<the post id from before>", "commentId": "<your copied comment id>"}
12. Grab the token from the browser and add it in Headers - Authorization: bearer <token>
13. SEND. (You have now updated the post with a comment)
14. Now GET to http://localhost:8080/posts/ (include token in header but no body)

The post should now have an array which contains your comment (populated with message and user object)

We have not tested this in jest. Added new ticket for this.

Please can we get opinions on this (should the posts/comments) be a separate controller/route, should CREATE and UPDATE be in the same method - for example.
